packages/lib-react-components/src/Markdownz/Markdownz.spec.js

```diff
@@ -1,14 +1,295 @@
 import React from 'react'
 import { shallow } from 'enzyme'
+import sinon from 'sinon'
+import { Image, Video } from 'grommet'
 import Markdownz from './Markdownz'
+import { jsx, markdown } from './helpers/testExamples'
 
-const content = '#Zooniverse'
-
+// TO DO: Add back working snapshots to test the overall HTML output
+// We have to use snapshots with styled-components becaue of the generated class names
 describe('<Markdownz />', function () {
   let wrapper
-  before(function () {
-    wrapper = shallow(<Markdownz>{content}</Markdownz>)
+
+  it('renders without crashing', function () { 
+    wrapper = shallow(<Markdownz>{markdown}</Markdownz>)
+    expect(wrapper).to.be.ok
+  })
+
+  it('parses markdown to jsx', function () {
+    wrapper = shallow(<Markdownz>{markdown}</Markdownz>)
+    expect(wrapper.equals(jsx)).to.be.true
+  })
+
+  it('correctly parses a user at-mention in a sentence', function () {
+    const sentence = `I would like to ping @am.zooni.`
```

Comment:
You can test your username regex more thoroughly by running this test for a set of usernames by looping through an array of test cases.
```javascript
describe('at-mentions', function () {
    ['team', 'am.zooni', 'a-user'].forEach(function (username) {
      describe(username, function () {
        it('should be recognised at the end of a sentence', function () {
          const sentence = `I would like to ping @${username}.`
          wrapper = shallow(<Markdownz>{sentence}</Markdownz>)
          const pingAnchor = wrapper.find({ href: `/users/${username}` })
          expect(pingAnchor).to.have.lengthOf(1)
          expect(pingAnchor.text()).to.equal(`@${username}`)
        })
      })
    })
 })
```

When I run that set of tests, `@am.zooni` and `@a-user` pass but `@team` fails. I'm not sure why.

Specs for the beginning and middle of a sentence should be tested too. I think the regex will catch them, but tests would check that.

Reply:
I'm an idiot. `@team` is listed in the code as an exception. `@srallen` passes. I'll push a commit.

Reply:
Resolved by 7953dcb.